Source/bmalloc/bmalloc/TZoneHeapManager.cpp

```diff
@@ -82,6 +95,38 @@ void TZoneHeapManager::init()
     if constexpr (verbose)
         TZONE_LOG_DEBUG("TZoneHeapManager initialization ");
 
+#if BUSE(BUCKETS_FOR_SIZE_CLASSES_FROM_ENVVAR)
+    // Allow the setting the bucket per size class params via an environment variable.
+    // TZONE_BUCKET_CONFIG=<bucket-for-small-sizes>:<buckets-for-large-sizes>:<max-small-size>
+    // Note that one, two or all three params can be given.
+    // For example, to specific 1 bucket for all size classes use TZONE_BUCKET_CONFIG=1:1
+    char* bucketsForSizeClassesValue = getenv("TZONE_BUCKET_CONFIG");
+    if (bucketsForSizeClassesValue && *bucketsForSizeClassesValue) {
+        constexpr unsigned numParams = 3;
+        constexpr unsigned bufferLength = 30;
+        char buffer[bufferLength + 1];
+        unsigned paramsAsNumbers[numParams];
+        unsigned paramsProvided = 0;
+
+        memcpy(buffer, bucketsForSizeClassesValue, bufferLength);
```

Comment:
what if `strlen(bucketsForSizeClassesValue) > bufferLength`? This will produce a non-nul-terminated `buffer`.

So then why bother copying bucketsForSizeClassesValue? It's not modified so seems you can just parse it in place and avoid this possibility.

Reply:
Yeah, I was thinking about that after posting.  I think if I check the length of the EnvVar and bail out, and drop a null in the last location, I'd be good.